src/main/java/com/varchenko/MikeHashMap.java

```diff
@@ -0,0 +1,225 @@
+package com.varchenko;
+
+
+import java.util.*;
+
+public class MikeHashMap<K, V> extends AbstractMap<K, V> implements Map<K, V> {
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int DEFAULT_CAPACITY = 16;
+    private Node<K, V>[] buckets;
+    private Node<K, V> current;
+    private int capacity;
+    private int size;
+    private int modCount;
+
+
+    public MikeHashMap() {
+        this.capacity = DEFAULT_CAPACITY;
+        buckets = new Node[capacity];
+    }
+
+    public MikeHashMap(int capacity) {
+        this();
+        this.capacity = capacity;
+    }
+
+    private void resize() {
+        if (size >= capacity * LOAD_FACTOR) {
+            Node<K, V>[] oldBucket = buckets;
+            buckets = new Node[capacity + (capacity >> 1)];
+            for (Node<K, V> kvNode : oldBucket) {
+                if (kvNode != null) {
+                    put(kvNode.getKey(), kvNode.getValue());
+                }
+            }
+        }
+    }
+
+    private int indexFor(Object k) {
+        if (k != null) {
+            int index = k.hashCode() & capacity - 1;
+            return index != 0 ? index : index + 1;
+        }
+        return 0;
+    }
+
+    private void putNotNull(K key, V value) {
+        Node<K, V> node = buckets[indexFor(key)];
+        if (node.getKey() != null) {
+            if (node.getKey().hashCode() == key.hashCode() && node.getKey().equals(key)) {
+                node.setValue(value);
+            } else {
+                node.setNext(new Node<>(key, value));
+                size++;
+                modCount++;
+            }
+        } else {
+            node.setValue(value);
+        }
+    }
+
+    private Node<K, V> findInNode(Node<K, V> node, Object k) {
```

Comment:
Rename method name to 'searchNodeByKey'. Use getBucket() method instead of passing 'node' params